projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3163,10 +3163,13 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
      */
     public hideOverlays() {
         this.overlayIDs.forEach(overlayID => {
-            if (this.overlayService.getOverlayById(overlayID)?.visible) {
+            const overlay = this.overlayService.getOverlayById(overlayID);
+
+            if (overlay?.visible && !overlay.closeAnimationPlayer?.hasStarted()) {
                 this.overlayService.hide(overlayID);
+
+                this.nativeElement.focus();
```

Comment:
if there is an overlay which is closing, `hasStarted == true`, and user starts to scroll this will not focus the native element. Not sure if this is an issue. Otherwise LGTM.

Reply:
We want to focus the native element only if closing the overlay is caused by the scroll. If closing has started before the scroll we shouldn't change the focus.